From 4ddb0c88b6b1cb9a6c180dab1651256ce5ed15f9 Mon Sep 17 00:00:00 2001
From: Michael Orlitzky <michael@orlitzky.com>
Date: Mon, 20 Jan 2025 22:55:08 -0500
Subject: [PATCH] m4rie.pc.in: improve compiler/linker flag handling

 1. List m4ri in Requires.private. This will automatically
    add -lm4ri for people who are statically linking, but will
    omit it when dynamically linking.

 2. Move $OPENMP_CFLAGS to Libs.private. Consumers of m4rie do not
    need to enable OpenMP at compile time to avoid problems, only at
    link time, and only when statically linking. In particular we want
    to _avoid_ enabling OpenMP by "accident" in a larger project that
    makes use of m4rie.

 3. Drop $M4RI_CFLAGS. There's just... no reason to add these here.
    In theory if M4RI had some important -D flags, they would go here,
    but at the moment it's just adding a bunch of -msse and -fopenmp
    stuff that we do not necessarily want to use when compiling
    consumers of m4rie.
---
 m4rie.pc.in | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/m4rie.pc.in b/m4rie.pc.in
index e716441..29bbc6d 100644
--- a/m4rie.pc.in
+++ b/m4rie.pc.in
@@ -6,5 +6,7 @@ includedir=@includedir@
 Name: M4RIE
 Description: Dense linear algebra over GF(2^e).
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lm4rie -lm4ri -lm
-Cflags: -I${includedir} @M4RI_CFLAGS@ @OPENMP_CFLAGS@
+Requires.private: m4ri >= 20240729
+Libs: -L${libdir} -lm4rie -lm
+Libs.private: @OPENMP_CFLAGS@
+Cflags: -I${includedir}
-- 
2.45.2

